src/java.base/share/classes/java/io/Reader.java

```diff
@@ -209,7 +209,8 @@ public int read(char[] cbuf, int off, int len) throws IOException {
             }
 
             public String readAllChars() throws IOException {
-                return cs.toString();
+                ensureOpen();
+                return cs.toString().substring(next);
```

Comment:
Your change implies creating a full-length string *first* (including cyoping and compression), just to strip it down to a smaller one *later*. It would be more efficient to *first* strip it down, and compression into a `String` *afterwards*. Imagine a huge `cs` with `next` being near to `length()`, and you see the difference in efficiency! 🙂 

`cs.subSequence(next, cs.length() - next).toString()`

Reply:
Oh, and we should set `next` afterwards, so the `Reader` knows that the end of the sequence is reached:

```java
public String readAllChars() throws IOException {
    ensureOpen();
    var remainder = cs.subSequence(next, cs.length() - next);
    next = length();
    return remainder.toString();
}
```

Reply:
So at least it's not just me... 😅 😉